### DISCUSSION TITLE
Admin panel

### DISCUSSION THREAD

**Discussion by rnbokade** [loco-rs/loco#577] [General]
Again coming form Django world, automatic admin panel generation, is a very lucrative feature. We should consider creating an admin panel. The point where i would say we could make it better than Django is,  We can include section that lists 
1. Worker jobs in queue
2. Number of instances and resources connected
3. List of tasks which could be run as worker job on single button click
along with the normal CRUD sections for different models.
We can also allow defining custom views which are admin specific within the codebase.
Another point we can differ the way we run admin panel could be running it as a separate service on different port rather than as a route within same service. e.g. `cargo loco admin` something of that sort... 
How this helps? It helps by giving a clear path of hiding admin from world. You just run admin on a server which is accessible to certain IPs over a certain network only by cleverly spinning it up on a server with proper firewall configs for doing so.

**Comment by jondot** [comment]
I’m wondering if you actually mean celery and it’s admin panel?

**Comment by rnbokade** [comment]
No I mean Django-Admin

**Comment by jondot** [comment]
got it. either way, try out one of the sidekiq admins, sidekiq-rs should be compatible, and give info about jobs, etc. as you described.
i believe the ability to mix-and-match admins is good, like what celery has.
since you described things related to background jobs I assume that is the focus.

on the other hand there is the rails admin project (there were multiple ones over the years actually), and that is an interesting approach by itself. it is a huge amount of work, so im willing to help anyone who decided to work on such as thing.
